sql/analyzer/optimization_rules.go

```diff
@@ -389,6 +389,16 @@ func evalFilter(ctx *sql.Context, a *Analyzer, node sql.Node) (sql.Node, error)
 					return e, nil
 				}
 
+				// UnaryMinus expressions come back from the parser when a negative float is evaluated. Treat them just like
+				// normal literal expressions.
+				if um, ok := e.(*expression.UnaryMinus); ok {
+					negated, err := e.Eval(ctx, nil)
```

Comment:
Can we have here the case that UnaryMinus is wrapping a non literal child?

Reply:
That should be covered by the isEvaluable check above (returns false if an expression contains columns). I don't want to assert a Literal child here because it wouldn't work for weird arithmetic expressions e.g. -(60*60*24).